python/ray/node.py

```diff
@@ -66,8 +66,8 @@ def __init__(self,
             self._register_shutdown_hooks()
 
         self.head = head
-        self.kernel_fate_share = (spawn_reaper
-                                  and ray.utils.detect_fate_sharing_support())
+        self.kernel_fate_share = bool(
+            spawn_reaper and ray.utils.detect_fate_sharing_support())
```

Comment:
Would it be better to do `spawn_reaper is not None and ...` to make this more clear?

Reply:
I think that wouldn't work, since `spawn_reaper` can be `False`?

Note that the `None` here wasn't coming from `spawn_reaper`, but from `detect_fate_sharing_support`. It currently returns `None` if there's no possible implementation for the platform at all, and it returns `False` if the platform could be supported but the functionality is somehow unavailable (e.g. permission issues or whatever).

Reply:
Oh - are the different return types both used? This seems pretty confusing. If we're just checking "supported or not" then why not just return `True/False`?

Reply:
It's not currently used in the code, but I generally try to avoid information loss like this, since it can be quite useful later for debugging. (e.g. if we see `None` I'd realize the code is either removed or otherwise not getting called, whereas if I see `False` I'd instantly know the it's probably a system-specific issue.) I can remove it if you feel it's overly confusing but if not I feel like it doesn't really do any harm.

Reply:
I think it does some amount of harm based on the fact that this PR is fixing a bug due to the confusion from it :) better to add this sort of distinction only when it's needed IMO

Reply:
Haha, well this didn't really cause a bug. `None` was still (correctly) treated as falsy. Rather, the warning mistriggered. I think you were probably right to suggest the argument should be mandatory though, let me change that.